### PR TITLE
updated threatmodel using supervisor domains terms 

### PR DESCRIPTION
**Pull request by rsahita**
updated threatmodel using supervisor domains terms 
-also added normative or non-normative refs to mitigations

### CODE REVIEW

**Review by sameo — APPROVED** [review]

File: specification/threatmodel.adoc [hunk 1]
```diff
@@ -1,187 +1,296 @@
...
+TEE memory | Sv, Smmtt
 
-| Memory Confidentiality | Memory isolation (read)     | Required                | cryptography and/or MMU, MPU, PMA/MTT extension | Prevent untrusted components from reading TEE memory       | AP-TEE TG specify 
-| Memory Confidentiality | Cipher text read prevention | Required                | cryptography and/or MMU, MPU, PMA/MTT extension | Prevent untrusted code from accessing encrypted TEE memory | AP-TEE TG specify 
-| Memory Confidentiality | Per TEE encryption          | Implementation-specific | cryptography and/or MMU, MPU, PMA/MTT extension | Each VM has one or more unique keys                        | AP-TEE TG to recommend 
-| Memory Confidentiality | Memory encryption strength  | Implementation-specific | cryptography                                    | Encryption algorithm and key strength                      | AP-TEE TG to recommend 
-| Memory Confidentiality | Number of encryption keys   | Implementation-specific | cryptography                                    | Number of TEE keys supported                               | AP-TEE TG to recommend 
+| Memory Confidentiality | Cipher text read prevention | Required | cryptography
```
Comment (NicholasWoodIMG) [inline specification/threatmodel.adoc:145]: > Cipher text read prevention 

I agree "prevent unauthorized access to memory assigned to a different domain/TVM, regardless of memory protection" is a sensible isolation guarantee.

But in a memory protection context it's about freshness guarantees. 

Maybe rephrase in terms of freshness requirements? Boot freshness; domain freshness; TVM freshness; temporal freshness; etc.?

And then do the same for replay protection, integrity protection.

And build a simple threat/mitigation table around that. 

Actual implementations will have different threat models for what is in/out of scope - software attacks, physical attacks, whatever - and what performance/security trade-offs they accepts.

**Review by NicholasWoodIMG — COMMENTED** [review]
Just some high-level comments on structure. I realise it's early days.

File: specification/threatmodel.adoc [hunk 2]
```diff
@@ -1,187 +1,296 @@
...
 debug mode accessible information.
 
 T21: A TVM causes a denial of service on the platform
 
 [NOTE]
 ====
-This is not an exhaustive list and will be updated on a regular basis as attacks evolve._
+This threat model is not an exhaustive list and will be updated (via the RISC-V
+Security Model specification <<R21>>) on a regular basis as attacks evolve.
 ====
```
[the remark this answers is not in the document]
Comment (rsahita) [reply specification/threatmodel.adoc:114]: right - this is a top level threat summary.
